File: matiasmasso/Mat.NET/DTO/DTORecollida.cs
using System;
using System.Collections.Generic;

namespace DTO
{
    public class DTORecollida
    {
        public int Id { get; set; }
        public DateTime Fch { get; set; } = DateTime.MinValue;
        public DTODelivery Delivery { get; set; }
        public string OrigenNom { get; set; }
        public string OrigenAdr { get; set; }
        public DTOZip OrigenZip { get; set; }
        public string OrigenTel { get; set; }
        public string OrigenContact { get; set; }
        public string DestiNom { get; set; }
        public string DestiAdr { get; set; }
        public DTOZip DestiZip { get; set; }
        public string DestiTel { get; set; }
        public string DestiContact { get; set; }
        public int Bultos { get; set; }
        public EstatsDeLaMercancia EstatDeLaMercancia { get; set; } = EstatsDeLaMercancia.Desconegut;
        public string Motiu { get; set; }
        public Carrecs Carrec { get; set; }
        public Accions Accio { get; set; }
        public List<DTOQtySku> Items { get; set; }

        public enum EstatsDeLaMercancia
        {
            Desconegut,
            Bo,
            Malmes
        }

        public enum Carrecs
        {
            Indeterminat,
            Nostre,
            Transportista,
            Magatzem
        }

        public enum Accions
        {
            Per_determinar,
            Entrar_en_stock,
            Tramitar_assegurança
        }


        public static DTORecollida Factory(DTODelivery oDelivery)
        {
            DTORecollida oRecollida = new DTORecollida();
            {
                var withBlock = oRecollida;
                withBlock.Delivery = oDelivery;
                withBlock.Fch = DTO.GlobalVariables.Today();
                withBlock.OrigenNom = oDelivery.Customer.NomComercialOrDefault();
                withBlock.OrigenAdr = oDelivery.Customer.Address.Text;
                withBlock.OrigenZip = oDelivery.Customer.Address.Zip;
                withBlock.OrigenTel = oDelivery.Tel;
                withBlock.DestiNom = oDelivery.Mgz.NomComercialOrDefault();
                DTOAddress oAddress = oDelivery.Mgz.Address;
                withBlock.DestiAdr = oAddress.Text;
                withBlock.DestiZip = oAddress.Zip;
                withBlock.DestiTel = oDelivery.Mgz.Telefon;
                withBlock.Items = new List<DTOQtySku>();
                foreach (var item in oDelivery.Items)
                    withBlock.Items.Add(new DTOQtySku(item.Qty, item.Sku));
            }
            return oRecollida;
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTORecollida.vb
Public Class DTORecollida
    Property Id As Integer
    Property Fch As Date = Date.MinValue
    Property Delivery As DTODelivery
    Property OrigenNom As String
    Property OrigenAdr As String
    Property OrigenZip As DTOZip
    Property OrigenTel As String
    Property OrigenContact As String
    Property DestiNom As String
    Property DestiAdr As String
    Property DestiZip As DTOZip
    Property DestiTel As String
    Property DestiContact As String
    Property Bultos As Integer
    Property EstatDeLaMercancia As EstatsDeLaMercancia = EstatsDeLaMercancia.Desconegut
    Property Motiu As String
    Property Carrec As Carrecs
    Property Accio As Accions
    Property Items As List(Of DTOQtySku)

    Public Enum EstatsDeLaMercancia
        Desconegut
        Bo
        Malmes
    End Enum

    Public Enum Carrecs
        Indeterminat
        Nostre
        Transportista
        Magatzem
    End Enum

    Public Enum Accions
        Per_determinar
        Entrar_en_stock
        Tramitar_assegurança
    End Enum


    Shared Function Factory(oDelivery As DTODelivery) As DTORecollida
        Dim oRecollida As New DTORecollida
        With oRecollida
            .Delivery = oDelivery
            .Fch = DateTime.Today
            .OrigenNom = oDelivery.Customer.NomComercialOrDefault()
            .OrigenAdr = oDelivery.Customer.Address.Text
            .OrigenZip = oDelivery.Customer.Address.Zip
            .OrigenTel = oDelivery.Tel
            .DestiNom = oDelivery.Mgz.NomComercialOrDefault()
            Dim oAddress As DTOAddress = oDelivery.Mgz.Address
            .DestiAdr = oAddress.Text
            .DestiZip = oAddress.Zip
            .DestiTel = oDelivery.Mgz.Telefon
            .Items = New List(Of DTOQtySku)
            For Each item In oDelivery.Items
                .Items.Add(New DTOQtySku(item.Qty, item.Sku))
            Next
        End With
        Return oRecollida
    End Function

End Class
